lib/pharos/phase.rb

```diff
@@ -61,5 +61,15 @@ def exec_script(script, vars = {})
     def parse_resource_file(path, vars = {})
       Pharos::YamlFile.new(resource_path(path)).read(vars)
     end
+
+    # @return [Hash]
+    def cluster_context
+      self.class.cluster_context
+    end
+
+    # @return [Hash]
+    def self.cluster_context
+      @@cluster_context ||= {} # rubocop:disable Style/ClassVars
```

Comment:
Should this be some sort of thread safe hash now that phases can execute in parallel?

Reply:
Probably, but it might be a bad idea anyway to write same key from multiple parallel phases?

Reply:
I've understood that ruby's standard `Hash` is safe for use in multiple MRI threads, so I'm not entirely sure what something like [`Concurrent::Hash`](http://ruby-concurrency.github.io/concurrent-ruby/Concurrent/Hash.html) actually does.

In terms of race conditions within our code, only one phase/thread should write to the `cluster_context`, and then following phases/threads can read from it. If we were to inappropriately change something like the `ConfigureMaster` to run in parallel, then the races would be in the actual  `copy_kube_certs`/`cache_kube_certs` methods, not in the hash accesses.

Reply:
https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent/hash.rb#L15-L16

To me it looks like that is all there is in `Concurrent::Hash` for MRI.

Reply:
I'd still change that to `@cluster_context` and change the instance-level accessor to point to `Pharos::Phase.cluster_context`, because why not. Class variables are weird.

Reply:
well, I wasn't really 100% sure if standard Hash was safe to use in many threads, but it seems to be. at least in our cases right now.

Reply:
We can do that stuff to the `Pharos::Kube::Session` (or some `Pharos::Session` thing) once this and #225 is in.